packages/dd-trace/src/appsec/blocking.js

```diff
@@ -5,13 +5,23 @@ const blockedTemplates = require('./blocked_templates')
 
 let templateHtml = blockedTemplates.html
 let templateJson = blockedTemplates.json
+let blockingConfiguration
 
-function block (req, res, rootSpan, abortController) {
-  if (res.headersSent) {
-    log.warn('Cannot send blocking response when headers have already been sent')
-    return
+function blockWithRedirect (res, rootSpan, abortController) {
+  rootSpan.addTags({
+    'appsec.blocked': 'true'
+  })
+
+  res.writeHead(blockingConfiguration.parameters.status_code, {
+    'Location': blockingConfiguration.parameters.location
+  }).end()
```

Comment:
Why don't you use `res.redirect()`?
```suggestion
  res.redirect(blockingConfiguration.parameters.status_code, blockingConfiguration.parameters.location).end()
```

Reply:
res.redirect is an express method, not available in nodejs http, res.writeHead is a http method.